### PR TITLE
Update issue and PR templates, add config, stale bot not

### LINKED ISSUE #55853: Update issue template to the latest in GitHub
### Description of Issue
Template is good, but a warning in GitHub to update to the latest and could include all new issues with a label such as `new` or `needs triage` to be excluded from the stale bot configuration.

### Setup
GitHub > SaltStack org > Salt repo > Code > github. > templates

### Steps to Reproduce Issue
see above

### Versions Report
N/A

### PR DESCRIPTION
**Pull request by sagetherage**
### What does this PR do?
Uses GitHub templates for [Bug] and [Feature Request] adds labels, easier to find by syntax search, adds syntax to PR template `Fixes:` to close issues linked with PRs as stated in the Contributing Guidelines, plus adds a config file with links for questions and discussions.

### What issues does this PR fix or reference?
Fixes: #55853

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Salt Community Slack
+    url: https://saltstackcommunity.slack.com/
+    about: Please ask and answer questions here.
+  - name: Salt-Users Forum
+    url: https://groups.google.com/forum/#!forum/salt-users
+    about: Please ask and answer questions here.
+     - name: Salt on Freenod
```
Comment (waynew) [inline .github/ISSUE_TEMPLATE/config.yml]:
```suggestion
     - name: Salt on Freenode
```

**Review by waynew — CHANGES_REQUESTED** [review]
Found a missing `e` :upside_down_face:

**Review by Ch3LL — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (sagetherage) [reply .github/ISSUE_TEMPLATE/config.yml]: @waynew ok all fixed thank you!

**Comment by max-arnold** [comment]
May I suggest adding the following snippet to the versions report section to make it collapsed by default?

    <details><summary>salt --versions-report</summary>
    
    ```
    PASTE HERE
    ```
    </details>

Another suggestion is to consider adding the `Docs & changelog written?` section to the PR template (in order to improve the docs coverage).

**Comment by bryceml** [comment]
https://github.com/saltstack/salt/pull/56075

could probably be taken care of on this pr and that one closed.

**Comment by sagetherage** [comment]
I guess I decided @dwoz when you review this for merging, please comment.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 
 ### Previous Behavior
 Remove this section if not relevant
 
 ### New Behavior
 Remove this section if not relevant
 
+### Docs & changelog written?
+Yes/No 
+
```
Comment (waynew) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Maybe do:

```suggestion
- [ ] Docs
- [ ] Changelog

```

**Review by waynew — CHANGES_REQUESTED** [review]
I think this will render the GFM checkbox?

**Review by waynew — APPROVED** [review]

**Pull request merged**